velox/docs/velox-in-10-min.rst

```diff
@@ -0,0 +1,321 @@
+If a function you need is not available, feel free to add a new one by following
+the :doc:`How to add a scalar function? <develop/scalar-functions>` guide.
+
+Check out :doc:`Expression Evaluation <develop/expression-evaluation>` article
+to learn what happens under the hood or play with `velox/examples/ExpressionEval.cpp
+<https://github.com/facebookincubator/velox/blob/main/velox/examples/ExpressionEval.cpp>`_
+which contains a detailed step-by-step example of constructing and evaluating
+expression trees using non-test library APIs.
+
+Queries
+-------
+
+Aggregations
+~~~~~~~~~~~~
+
+We can calculate sum and average of 'a' and 'b' by creating and
+executing a query plan with an aggregation node:
+
+.. code-block:: c++
+
+    auto plan = PlanBuilder()
+                  .values({data})
+                  .singleAggregation(
+                      {},
+                      {"sum(a) AS sum_a",
+                       "avg(a) AS avg_a",
+                       "sum(b) AS sum_b",
+                       "avg(b) AS avg_b"})
+                  .planNode();
+
+    auto sumAvg = getResults(plan);
+
+And here are the results:
+
+.. code-block:: c++
+
+    std::cout << sumAvg->toString() << std::endl;
+    std::cout << sumAvg->toString(0, sumAvg->size()) << std::endl;
+
+.. code-block::
+
+    > [ROW ROW<sum_a:BIGINT,avg_a:DOUBLE,sum_b:BIGINT,avg_b:DOUBLE>: 1 elements, no nulls]
+    0: {15, 2.5, 75, 12.5}
+
+You can use any of the available :doc:`Presto aggregate functions <functions/aggregate>`
+or add a new one by following :doc:`How to add an aggregate function? <develop/aggregate-functions>`
+guide. Check out :doc:`Aggregations <develop/aggregations>` article for a deep dive
+into aggregation-specific optimizations available in Velox.
+
+Sorting
+~~~~~~~
+
+We can sort data using the OrderBy plan node.
+
+.. code-block:: c++
+
+    plan = PlanBuilder().values({data}).orderBy({"a DESC"}, false).planNode();
+
+    auto sorted = getResults(plan);
+    std::cout << sorted->toString() << std::endl;
+    std::cout << sorted->toString(0, sorted->size()) << std::endl;
+
+.. code-block::
+
+    > [ROW ROW<a:BIGINT,b:BIGINT>: 6 elements, no nulls]
+    0: {5, 25}
+    1: {4, 20}
+    2: {3, 15}
+    3: {2, 10}
+    4: {1, 5}
+    5: {0, 0}
+
+And we can get top 3 rows using TopN node:
+
+.. code-block:: c++
+
+    plan = PlanBuilder().values({data}).topN({"a DESC"}, 3, false).planNode();
+
+    auto top3 = getResults(plan);
+    std::cout << top5->toString() << std::endl;
+    std::cout << top5->toString(0, top3->size()) << std::endl;
+
+.. code-block::
+
+    > [ROW ROW<a:BIGINT,b:BIGINT>: 3 elements, no nulls]
+    0: {5, 25}
+    1: {4, 20}
+    2: {3, 15}
+
+Connectors
+~~~~~~~~~~
+
+We have seen how to use Velox to perform computation on in-memory vectors
+provided by the caller. Velox can also pull data from connectors. There are
+two connectors to choose from. Hive connector reads ORC and Parquet
```

Comment:
We don't have ORC support. Does Meta talk about DWRF externally ?

Reply:
Let me change to DWRF.